packages/core-utils/src/itinerary.js

```diff
@@ -26,6 +26,23 @@ export function isTransit(mode) {
   return transitModes.includes(mode) || mode === "TRANSIT";
 }
 
+export function isOptional(leg) {
+  return leg.boardRule === "mustPhone";
+}
+export function isContinuousDropoff(leg) {
+  return leg.alightRule === "coordinateWithDriver";
+}
+export function isFlex(leg) {
+  return isOptional(leg) || isContinuousDropoff(leg);
+}
+
+export function bookingInfoDropofHasBookingTime(info) {
+  return info?.latestBookingTime?.daysPrior > 0;
+}
+export function legDropoffHasBookingTime(leg) {
```

Comment:
Maybe rename to `legDropoffRequiresAdvanceBooking` ?

Reply:
I love it! Changed in f7aec8908db009cc5d1b7ddbfe2e062539692f51